src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonInEnumerationCheck.java

```diff
@@ -0,0 +1,110 @@
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks if unnecessary semicolon is in enum definitions.
+ * Semicolon is not needed if enum body is empty.
+ * The check will flag the following with warnings:
+ * </p>
+ * <pre>
+ * enum One {
+ *     A,B; // violation
+ * }
+ * enum Two {
+ *     A,B,; // violation
+ * }
+ * enum Three {
+ *     A,B(); // violation
+ * }
+ * enum Four {
+ *     A,B{}; // violation
+ * }
+ * enum NoSemicolon {
+ *     A,B // OK
+ * }
+ * enum Normal {
+ *     A,
+ *     B
+ *     ; // required ";", no violation
+ *     Normal(){}
+ * }
+ * </pre>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;UnnecessarySemicolonInEnumeration&quot;/&gt;
+ * </pre>
+ *
+ * @since 8.22
+ */
+@StatelessCheck
+public final class UnnecessarySemicolonInEnumerationCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_SEMI = "unnecessary.semicolon";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.ENUM_CONSTANT_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST semicolon = findSemicolonSibling(ast);
+        if (semicolon != null && hasNoCodeElementsAfter(semicolon)) {
+            log(semicolon, MSG_SEMI);
+        }
+    }
+
+    private static DetailAST findSemicolonSibling(DetailAST ast) {
+        DetailAST result = ast;
+        do {
+            result = result.getNextSibling();
+        } while (result != null && result.getType() != TokenTypes.SEMI);
+        return result;
+    }
+
+    private static boolean hasNoCodeElementsAfter(DetailAST ast) {
```

Comment:
Since it is a boolean result, isn't it preferred to start with `is` and be in the positive tense?
What about `isEndOfEnumerationAfter`?

Reply:
done.